### PR TITLE
[MB-1131]Combined methods create queue and add permission

### PR DESCRIPTION
**Pull request by sdkottegoda**
Addresses the issue reported at https://wso2.org/jira/browse/MB-1131

Please merge https://github.com/wso2/andes/pull/351 along with this.

### CODE REVIEW

File: components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java [hunk 1]
```diff
@@ -553,6 +546,34 @@ public void updatePermission(String queueName, QueueRolePermission[] queueRolePe
...
+     * @param queueRolePermissionsDTO A {@link org.wso2.carbon.andes.admin.internal.QueueRolePermission}
+     * @throws BrokerManagerAdminException
+     */
+    public void addQueueAndAssignPermission(String queueName, QueueRolePermission[] queueRolePermissionsDTO)
+            throws BrokerManagerAdminException {
+        QueueManagerService queueManagerService = AndesBrokerManagerAdminServiceDSHolder.getInstance()
+                .getQueueManagerService();
+        org.wso2.carbon.andes.core.types.QueueRolePermission[] rolePermissions;
+        try {
+            if (queueRolePermissionsDTO != null && queueRolePermissionsDTO.length > 0) {
```
Comment (ramith) [inline components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java]: shall we flip?

File: components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/internal/QueueRolePermission.java [hunk 2]
```diff
@@ -47,4 +47,9 @@ public boolean isAllowedToPublish() {
     public void setAllowedToPublish(boolean isAllowedToPublish) {
         this.isAllowedToPublish = isAllowedToPublish;
     }
+
```
Comment (ramith) [inline components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/internal/QueueRolePermission.java]: method comments?

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java [hunk 3]
```diff
@@ -338,6 +338,27 @@ public void updatePermission(String queueName, QueueRolePermission[] queueRolePe
...
+        if (!QueueManagementBeans.queueExists(tenantBasedQueueName)) {
+
+            // create a queue with the provided name and assign permissions for the current user as well as for the
+            // explicitly provided roles
+            createQueue(tenantBasedQueueName);
+            updatePermission(tenantBasedQueueName, queueRolePermissions);
+
+        } else {
+            throw new QueueManagerException("Queue with the name: " + queueName + " already exists!",
+                    new RuntimeException("Queue with the name: " + queueName + " already exists!"));
```
Comment (ramith) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java]: this looks funky! shall we re-visit?

File: components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java (hunk 1 above)
Comment (hemikak) [inline components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java]: +1

File: components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java [hunk 4]
```diff
@@ -531,19 +531,12 @@ public void updatePermission(String queueName, QueueRolePermission[] queueRolePe
...
-                    permission.setAllowedToConsume(queueRolePermission.isAllowedToConsume());
-                    permission.setAllowedToPublish(queueRolePermission.isAllowedToPublish());
-                    permissionList.add(permission);
-                }
+
                 rolePermissions =
-                    new org.wso2.carbon.andes.core.types.QueueRolePermission[permissionList.size()];
-                permissionList.toArray(rolePermissions);
+                        new org.wso2.carbon.andes.core.types.QueueRolePermission[queueRolePermissionsDTO.length];
+                for (int i=0; i < queueRolePermissionsDTO.length; i++) {
```
Comment (hemikak) [inline components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java]: Issue with readability. Please keep a spaces between i=0.

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java (hunk 3 above)
Comment (hemikak) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java]: Wont throwing a RuntimeException cause the broker to break?

File: components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java (hunk 1 above)
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/AndesAdminService.java]: flipped

File: components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/internal/QueueRolePermission.java (hunk 2 above)
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.admin/src/main/java/org/wso2/carbon/andes/admin/internal/QueueRolePermission.java]: Added comments

File: components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java (hunk 3 above)
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java]: We don't need to pass a runtime exception to the constructor of the QueueManagerException. Corrected that.

**Pull request merged**